src/npc.cpp

```diff
@@ -439,9 +439,17 @@ void npc::randomize_from_faction( faction *fac )
 
 void npc::set_fac( const string_id<faction> &id )
 {
+    if( my_fac ) {
+        my_fac->remove_member( getID() );
+    }
     my_fac = g->faction_manager_ptr->get( id );
-    fac_id = my_fac->id;
-    my_fac->add_to_membership( getID(), disp_name(), known_to_u );
+    if( my_fac ) {
+        my_fac->add_to_membership( getID(), disp_name(), known_to_u );
+        fac_id = my_fac->id;
+    } else {
+        my_fac = nullptr;
```

Comment:
Redundant assignment. If `my_fac` evaluates to `false` in the if-condition above, it means it is null. Setting it here has no effect.